packages/dnd/src/find-target-element/find-target-element.ts

```diff
@@ -0,0 +1,61 @@
+const findTargetElement = (position: any, state: any): any => {
+    let result = null;
+    let resultContainer = null;
+    let resultPosition;
+
+    let margin = Infinity;
+
+    // find most suitable container
+    Object.keys(state.containers).forEach(key => {
+        let temporaryResult = null;
+
+        const container = state.containers[key];
+
+        const metadata = container.ref.getBoundingClientRect().toJSON();
+
+        // check if mouse is inside a container
+        if (
+            position.x >= metadata.x
+            && position.x <= metadata.right
+            && position.y >= metadata.y
+            && position.y <= metadata.bottom
+            && position.x - metadata.left <= margin
+        ) {
+            // check if users wants to move item under another item
+            container.children.forEach((id: any, index: number) => {
```

Comment:
You might want to consider the traditional `for` loop. It should give you better performance. There will be a lot of looping around and computations. I would not suggest it otherwise.